api/server/router/build/build_routes.go

```diff
@@ -72,11 +72,13 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 	options.RemoteContext = r.FormValue("remote")
 	if versions.GreaterThanOrEqualTo(version, "1.32") {
 		apiPlatform := r.FormValue("platform")
-		p := system.ParsePlatform(apiPlatform)
-		if err := system.ValidatePlatform(p); err != nil {
-			return nil, errdefs.InvalidParameter(errors.Errorf("invalid platform: %s", err))
+		if len(strings.TrimSpace(apiPlatform)) != 0 {
```

Comment:
Where is this (`strings.TrimSpace()`) coming from? It seems you are relaxing the spec here, allowing for non-empty strings containing some whitespace characters. Previous implementation (`system.ParsePlatform()`) did not do any special treatment of white space.